extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,1074 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
     width="480" height="270"></amp-youtube>
 <figcaption>Watch this video for an introduction to amp-bind.</figcaption></figure>
 
-### A simple example
+`amp-bind` has three main concepts:
+
+1. [State](#state): A document-scope, mutable JSON state. State variables update in response to user actions. `amp-bind` does not evaluate expressions on page load. Visual elements should have their default "state" defined and not rely `amp-bind` for initial render.
+2. [Expressions](#expressions): JavaScript-like expressions that can reference the **state**.
+3. [Bindings](#bindings): Special attributes that link an element's property to an **expression**. A property is bound by wrapping it inside brackets, in the form of `[property]`.
 
-In the following example, tapping the button changes the `<p>` element's text from "Hello World" to "Hello amp-bind".
+### Basic example
+
+[example preview="inline" playground="true" imports="amp-bind"]
 
 ```html
 <p [text]="'Hello ' + foo">Hello World</p>
 
-<button on="tap:AMP.setState({foo: 'amp-bind'})">Say "Hello amp-bind"</button>
+<button on="tap:AMP.setState({foo: 'Interactivity'})">
+  Say "Hello Interactivity"
+</button>
 ```
 
-{% call callout('Note', type='note') %}
-For performance and to avoid the risk of unexpected content jumping, `amp-bind` does not evaluate expressions on page load. This means that the visual elements should be given a default state and not rely `amp-bind` for initial render.
-{% endcall %}
-
-### How does it work?
+[/example]
 
-`amp-bind` has three main components:
+In the example above:
 
-1. [State](#state): A document-scope, mutable JSON state. In the example above, the state is empty before tapping the button. After tapping the button, the state is `{foo: 'amp-bind'}`.
-2. [Expressions](#expressions): These are JavaScript-like expressions that can reference the **state**. The example above has a single expression, `'Hello ' + foo`, which concatenates the string literal `'Hello '` and the state variable `foo`.
-   There is a limit of 100 operands what can be used in an expression.
-3. [Bindings](#bindings): These are special attributes of the form `[property]` that link an element's property to an **expression**. The example above has a single binding, `[text]`, which updates the `<p>` element's text every time the expression's value changes.
+- The **state** begins as empty.
+- It has a single **binding** to `[text]`, the text content of a node, on the `<p>` element.
+- The `[text]` value contains the **expression**, `'Hello ' + foo`.
+  - This expression concatenates the string 'Hello ' and the value of the **state variable** foo.
+- When the user taps/clicks the button, it triggers the `tap` event.
```

Comment:
Suggestion: turn this into an enumerated list

```
When the user taps/clicks the button:

1. It triggers the `tap` event.
2. The `tap` event invokes the `AMP.setState()` method.
3. The `AMP.setState()` methods sets the `foo` **state variable** to the value of `Interactivity`.
4. The state is no longer empty, so the pages updates the bound property to its state.
```

Reply:
Great explanation btw!

Reply:
ty, ty!